cpp/oneapi/dal/table/backend/homogen_kernels.cpp

```diff
@@ -130,33 +130,20 @@ static void pull_row_major_impl(const Policy& policy,
 
         auto src_data = origin_data.get_data() + origin_offset * origin_dtype_size;
         auto dst_data = block_data.get_mutable_data();
-
-        if (block_info.get_column_count() > 1) {
```

Comment:
Just because I am not familiar, it looks like just removing the if and relying on the for loop (a simplification here).